scripts/linters/js_ts_linter.py

```diff
@@ -429,6 +433,74 @@ def _check_angular_services_index(self) -> concurrent_task_utils.TaskResult:
         return concurrent_task_utils.TaskResult(
             name, failed, error_messages, error_messages)
 
+    def _check_unknown_type(self) -> concurrent_task_utils.TaskResult:
+        """Prints a list of lint errors if an unknown type is used. This lint
+            check is not enabled by default. Add proper comment if unknown is
+            needed.
+
+        Returns:
+            TaskResult. A TaskResult object representing the result of the
+            lint check.
+        """
+        name = 'Unknown type'
+        error_messages = []
+        failed = False
+        comment_before_unknown_type = False
+        ts_files_to_check = self.ts_filepaths
+        for file_path in ts_files_to_check:
+            # Not showing lint errors for files present in
+            # FILES_CONTAIN_UNKNOWN_TYPE
+            if file_path in FILES_CONTAINING_UNKNOWN_TYPE:
+                continue
+
+            file_content = self.file_cache.read(file_path)
+            for line_num, line in enumerate(file_content.split('\n')):
+                # Indexes where unknown type (: unknown) is present in a 
+                # particular line.
+                unknown_type_object = (
+                    re.finditer(pattern=': unknown', string=line))
+                unknown_type = (
+                    [index.start() for index in unknown_type_object])
+                # Indexes where unknown type conversion (as unknown) is
+                # present in a particular line.
+                unknown_type_conversion_object = (
+                    re.finditer(pattern='as unknown', string=line))
+                unknown_type_conversion = (
+                    [index.start() for index in unknown_type_conversion_object])
+
+                # Checking previous line contain comment, if yes then skip throw
+                # errors.
+                if not comment_before_unknown_type:
+                    # Throw error if unknown type is present.
+                    if len(unknown_type):
+                        failed = True
+                        for x in range(len(unknown_type)):
+                            error_message = (
+                                '%s:%s:%s: unknown type used. Add proper comment if'
+                                ' Unknown is needed.' % (
+                                file_path, line_num + 1, unknown_type[x]))
+                            error_messages.append(error_message)
+
+                    # Throw error if unknown type conversion is present.
+                    if len(unknown_type_conversion):
+                        failed = True
+                        for x in range(len(unknown_type_conversion)):
+                            error_message = (
+                                '%s:%s:%s: unknown type conversion used. Add proper'
```

Comment:
I think the word "unknown" is confusing here since it could mean either "a type conversion of kind `unknown`" or "a type conversion that the linter doesn't know about". Can you re-word to clarify? Maybe just add backticks around "unknown" so that it's clearly a code reference

Reply:
not done

Reply:
done